packages/menu/src/MenuItem.tsx

```diff
@@ -137,43 +137,49 @@ const disbaledTextStyle = css`
   color: ${uiColors.gray.light1};
 `;
 
-export interface MenuItemProps {
-  /**
-   * If supplied, MenuItem will be rendered inside of `a` tags.
-   */
-  href?: string;
-
+interface SharedMenuItemProps {
   /**
    * Class name that will be applied to root MenuItem element.
    */
   className?: string;
 
   /**
-   * Callback function to be executed when MenuItem is clicked.
+   * Main content displayed in MenuItem.
    */
-  onClick?: React.MouseEventHandler;
+  children?: React.ReactNode;
 
   /**
-   * Main content displayed in MenuItem.
+   * Determines whether or not the MenuItem is active.
    */
-  children?: React.ReactNode;
+  active?: boolean;
 
   /**
    * Description text displayed below title in MenuItem.
    */
   description?: string;
-
   /**
    * Determines whether or not the MenuItem is disabled.
    */
   disabled?: boolean;
+  ref?: any;
+}
 
-  /**
-   * Determines whether or not the MenuItem is active.
-   */
-  active?: boolean;
+interface LinkMenuItemProps extends HTMLElementProps<'a'>, SharedMenuItemProps {
+  href: string;
+}
 
-  ref?: any;
+interface ButtonMenuItemProps
+  extends HTMLElementProps<'button'>,
+    SharedMenuItemProps {
+  href?: null;
+}
+
+type MenuItemProps = LinkMenuItemProps | ButtonMenuItemProps;
+
+function usesLinkElement(
+  props: LinkMenuItemProps | ButtonMenuItemProps,
+): props is LinkMenuItemProps {
+  return props.href != null;
```

Comment:
I know we've talked about whether we should write `value != null` or `value !== undefined && value !== null`...

@hswolff @brookescarlett Do we want to take a stance on this and prefer `value != null` in the future? In the past, I think Harry and I spoke about this and preferred the more verbose option in the past for readability, but I'd personally be fine with this instead.

Reply:
Personally support the less verbose option, but will move forward based on whatever you think.

Reply:
FWIW at least on MMS we don't have a standard. If anything, I run into `value != null` more often, and we have gone as far as configuring ESLint to allow that as an exception to `eqeqeq`.

Reply:
I'm completely fine with `!= null` shortcut.

Reply:
💯